### ISSUE TITLE
Scatter (scatter.js) test for the use of logarithmic scale relies on minified d3.js

### ISSUE THREAD

**Issue by mlawry** [novus/nvd3#1520]
In the file [scatter.js](https://github.com/novus/nvd3/blob/master/src/models/scatter.js#L102) on line 102:

    var logScale = chart.yScale().name === d3.scale.log().name ? true : false;

`chart.yScale()` returns a function and so do `d3.scale.log()`. Testing on the `name` property will return the same string value "scale" because that's the name of the function d3.scale.log() returns (see [d3/log.js](https://github.com/mbostock/d3/blob/v3.5.2/src/scale/log.js#L20)). This means the test will always be true, which is probably unintended.

In Internet Explorer 11, the `name` property of functions is `undefined`, which means the test will always be true.

The test does work if I use the minified d3.min.js in Chrome, because the minification process usually renames the "scale" inner functions within `d3.scale.linear()` and `d3.scale.log()` to different letters (e.g. "i" and "o").

**Comment by mgalloue** [comment]
Yep, totally true. We need to find another way to check the scale type :-/ . Any suggestions?

**Comment by liquidpele** [comment]
https://stackoverflow.com/questions/16841010/how-to-get-the-type-of-a-scale-in-d3  

maybe?

**Comment by mlawry** [comment]
Why not just submit a pull request to mbostock/d3 that adds a `scaleType` property to d3.scales and hope they accept? :-)

Otherwise, without touching d3, I suggest the lazy method of adding a `useLogScale` option to scatter.js. But anyway, I'll let you experts decide.

**Comment by bernhard-42** [comment]
I ran into the same issue and looked into d3-scale. It looks like the log scale is the only scale having a base() method (for selecting the appropriate log base)
So, wow about changing the logScale check to

`
var logScale = ((typeof(chart.yScale().base) == "function")
`

**Comment by liquidpele** [comment]
If you find a good way to fix this, send me a pull request... just make sure the change doesn't break NON-log scales :p

**Comment by bernhard-42** [comment]
please see https://github.com/novus/nvd3/pull/2012
